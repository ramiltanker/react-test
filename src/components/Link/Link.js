import React from "react";

import { useDispatch } from "react-redux";

import { ADD_TODO } from "../../services/constants/constants";

function Link(props) {
  const { title, url } = props;

  const dispatch = useDispatch();

  const onClick = (e) => {
    e.preventDefault();
    alert("Ура! Пельмени!");
    dispatch({ type: ADD_TODO, text: "пельмени" });
  };

  return (
    <a href={url} onClick={(e) => onClick(e)}>
      {title}
    </a>
  );
}

export default Link;
