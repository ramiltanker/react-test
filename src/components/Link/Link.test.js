import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import { testRender, fixtureSet, makeTestStore } from "../../services/testUtils";

import Link from "../Link/Link";

it("Ссылка рендерится без ошибок", () => {
  const tree = renderer
    .create(<Link title="Рецепт пельменей" url="#" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Нажатие на кнопку вызывает корректный alert", () => {
  window.alert = jest.fn();

  // Рендерим ссылку в переменную
  const output = shallow(
    <Link title="Рецепт пельменей" url="https://pelmeni.gov" />
  );
  // Имитируем нажатие на ссылку
  output.simulate("click");

  // Проверяем, что alert сработал с правильным текстом предупреждения
  expect(window.alert).toHaveBeenCalledWith("Ура! Пельмени!");
});
