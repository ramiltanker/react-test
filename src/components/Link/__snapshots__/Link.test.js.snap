// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ссылка рендерится без ошибок 1`] = `
<a
  href="https://pelmeni.gov"
  onClick={[Function]}
>
  Рецепт пельменей
</a>
`;
